File: src/main/java/us/fatehi/palindrome/Palindrome.java
package us.fatehi.palindrome;


public class Palindrome
{

  public boolean isPalindrome(final String word)
  {
    if (word == null || word.length() < 2)
    {
      return true;
    }
    else
    {
      final char firstChar = word.charAt(0);
      final char lastChar = word.charAt(word.length() - 1);
      final String center = word.substring(1, word.length() - 1);
      return (firstChar == lastChar) && isPalindrome(center);
    }
  }

}

File: C#/Palindrome/Palindrome.cs
namespace us.fatehi.palindrome
{
    public class Palindrome
    {
        public bool IsPalindrome(string word)
        {
            // Exit early for trivial cases, such as null,
            // empty string, and a word with a single letter
            if (word == null || word.Length < 2)
            {
                return true;
            }

            var firstChar = word[0];
            var lastChar = word[^1];
            var center = word[1..^1];
            return (firstChar == lastChar) && IsPalindrome(center);
        }
    }
}
